public/app/features/explore/TraceView/createSpanLink.tsx

```diff
@@ -80,39 +84,48 @@ function legacyCreateSpanLinkFactory(splitOpenFn: SplitOpen, traceToLogsOptions?
     // the moment. Issue is that the trace itself isn't clearly mapped to dataFrame (right now it's just a json blob
     // inside a single field) so the dataLinks as config of that dataFrame abstraction breaks down a bit and we do
     // it manually here instead of leaving it for the data source to supply the config.
+    let dataLink: DataLink<LokiQuery | DataQuery> = {} as DataLink<LokiQuery | DataQuery>;
+    let link: LinkModel<Field>;
 
-    const expr = getLokiQueryFromSpan(span, traceToLogsOptions);
-    if (!expr) {
+    /** getLinkForLoki will return undefined if there are no tags in the datasource config;
+     * in that case we skip building a dataLink below and return undefined for SpanLink function
+     */
+    let lokiQuery = getLinkForLoki(span, traceToLogsOptions, dataSourceSettings);
```

Comment:
This still unnecessarily creates a Loki query for a Splunk datasource. I was thinking something along the lines of:
```
switch (dataSourceSettings?.type) {
  case 'loki':
    dataLink = getLinkForLoki(span, ...);
    if (!dataLink) {
      return undefined
    }
  case 'grafana-splunk-datasource':
    ...
```

In that case, we also only need to return the datalink from each function instead of the query/expr and datalink